### PR TITLE
Zfinx: Standard Extension for floating point integer registers

### PR DESCRIPTION
**Pull request by rnbwdsh**
TL:DR: Use regular int registers for float instructions instead of the float_regs.

* Moving all float-instructions from iss into inline-functions in iss_float.h
* Add RegFile::write_f32/read_f32 similar to to Regfile.read/write or fp_regs 
* Move some defines (RAISE_ILLEGAL_INSTRUCTION, REQUIRE_ISA, RD, RS1, RS2, RS3) to header file 
* Create a copy of iss_float.h: iss_zfinx.h
* Replace the fp_regs.write -> regs.write_f32 in zfinx
* Replace the fp_regs.f3 -> regs.read_f32 in zfinx
* Set make zfinx forbidden instructions always RAISE_ILLEGAL_INSTRUCTION (covered via regular int instructions)

Used [reference manual](https://github.com/riscv/riscv-zfinx/blob/main/zfinx-1.0.0-rc.pdf)

### CODE REVIEW

**Comment by rnbwdsh** [comment]
Question 1: Do we always need the 
```
	fp_prepare_instr();
	fp_setup_rm();
        ...
        fp_finish_instr();
```
Stuff for zfinx? The manual states, misa.[F|D|Q] and status.fs are hardwired to 0 -> maybe no?

**Comment by rnbwdsh** [comment]
Question 2: Testing?

Probably needs to be adapted to run once with and once without zfinx defined.
